test/integration/tcpingress_test.go

```diff
@@ -23,8 +24,19 @@ import (
 	"github.com/kong/kubernetes-ingress-controller/pkg/clientset"
 )
 
+var tcpMutex sync.Mutex
+
 func TestTCPIngressEssentials(t *testing.T) {
 	t.Parallel()
+	// Ensure no other TCP tests run concurrently to avoid fights over the port
+	// Free it when done
+	t.Log("locking TCP port")
+	tcpMutex.Lock()
+	defer func() {
```

Comment:
```suggestion
	defer tcpMutex.UnLock()
```
replace the whole defer function,

Reply:
Keeping these as-is to preserve the grouped test logs and since the suggestions would create errors (they don't replace the entire function and use `UnLock` where they should use `Unlock`).

Reply:
Don't see any errors using generic/concise syntax
mutx.Lock
defer mutx.Unlock

un-nessary func() objects added extra load into the stack, not useful and redundant. 
Suggest to keep as concise as possible.

Reply:
You'd need to have changed multiple lines in the suggestion to make it valid. With the original proposal, it results in a syntax error from the leftover anonymous function content:
```
14:08:51-0700 yagody $ git diff
diff --git a/test/integration/tcpingress_test.go b/test/integration/tcpingress_test.go
index ebbe2937..97007723 100644
--- a/test/integration/tcpingress_test.go
+++ b/test/integration/tcpingress_test.go
@@ -32,7 +32,7 @@ func TestTCPIngressEssentials(t *testing.T) {
        // Free it when done
        t.Log("locking TCP port")
        tcpMutex.Lock()
-       defer func() {
+       defer tcpMutex.UnLock()
                t.Log("unlocking TCP port")
                tcpMutex.Unlock()
        }()
14:08:52-0700 yagody $ GOFLAGS="-tags=integration_tests" go vet ./...
test/integration/tcpingress_test.go:38:3: expected ';', found '('
```

Fixing that results in an undefined error:
```
14:09:16-0700 yagody $ git diff                                      
diff --git a/test/integration/tcpingress_test.go b/test/integration/tcpingress_test.go
index ebbe2937..4faf4969 100644
--- a/test/integration/tcpingress_test.go
+++ b/test/integration/tcpingress_test.go
@@ -32,10 +32,7 @@ func TestTCPIngressEssentials(t *testing.T) {
        // Free it when done
        t.Log("locking TCP port")
        tcpMutex.Lock()
-       defer func() {
-               t.Log("unlocking TCP port")
-               tcpMutex.Unlock()
-       }()
+       defer tcpMutex.UnLock()
 
        ns, cleanup := namespace(t)
        defer cleanup()
14:09:32-0700 yagody $ GOFLAGS="-tags=integration_tests" go vet ./...
# github.com/kong/kubernetes-ingress-controller/test/integration
vet: test/integration/tcpingress_test.go:35:17: tcpMutex.UnLock undefined (type sync.Mutex has no field or method UnLock)
```
Grouping these with log lines inside a function was intentional: it makes it clear when they're  released in case you need to debug an unexpected deadlock scenario. The extra stack depth I'm not too concerned with between it being constant and part of test code rather something in a critical path.

Reply:
Sorry making you confusing, I mean we should intentionally implement the code as following 
tcpMutex.Lock()
defer tcpMutex.UnLock()

removing following
> defer func() {
> -               t.Log("unlocking TCP port")
> -               tcpMutex.Unlock()
> -       }()
> 
>